File: src/Api/Model/Languageforge/Translate/TranslateRoles.php
<?php

namespace Api\Model\Languageforge\Translate;

use Api\Model\Shared\Rights\Domain;
use Api\Model\Shared\Rights\Operation;
use Api\Model\Shared\Rights\ProjectRoles;

class TranslateRoles extends ProjectRoles
{
    public static function init()
    {
        // Contributor
        $rights = [];
        $rights[] = Domain::PROJECTS + Operation::VIEW;
        $rights[] = Domain::ENTRIES + Operation::VIEW;
        $rights[] = Domain::ENTRIES + Operation::EDIT;
        $rights[] = Domain::ENTRIES + Operation::CREATE;
        $rights[] = Domain::ENTRIES + Operation::DELETE;
        self::$_rights[self::CONTRIBUTOR] = $rights;

        // Manager (everything a Contributor has... plus the following)
        $rights = self::$_rights[self::CONTRIBUTOR];
        $rights[] = Domain::PROJECTS + Operation::EDIT;
        $rights[] = Domain::USERS + Operation::CREATE;
        $rights[] = Domain::USERS + Operation::EDIT;
        $rights[] = Domain::USERS + Operation::DELETE;
        $rights[] = Domain::USERS + Operation::VIEW;
        self::grantAllOnDomain($rights, Domain::ENTRIES);
        self::$_rights[self::MANAGER] = $rights;
    }

    private static $_rights;
    public static function hasRight($role, $right) { return self::_hasRight(self::$_rights, $role, $right); }
    public static function getRightsArray($role) { return self::_getRightsArray(self::$_rights, $role); }
}

TranslateRoles::init();

File: src/netcore-api/SIL.XForge.WebApi.Server/Models/Translate/TranslateRoles.cs
using System.Collections.Generic;

namespace SIL.XForge.WebApi.Server.Models.Translate
{
    public class TranslateRoles : ProjectRoles
    {
        public static TranslateRoles Instance { get; } = new TranslateRoles();

        private TranslateRoles()
        {
            var contributorRights = new HashSet<Right>
            {
                new Right(Domain.Projects, Operation.View),
                new Right(Domain.Entries, Operation.View),
                new Right(Domain.Entries, Operation.Edit),
                new Right(Domain.Entries, Operation.Create),
                new Right(Domain.Entries, Operation.Delete)
            };
            Rights[Contributor] = contributorRights;

            var managerRights = new HashSet<Right>(contributorRights)
            {
                new Right(Domain.Projects, Operation.Edit),
                new Right(Domain.Users, Operation.Create),
                new Right(Domain.Users, Operation.Edit),
                new Right(Domain.Users, Operation.Delete),
                new Right(Domain.Users, Operation.View)
            };
            managerRights.UnionWith(AllRights(Domain.Entries));
            Rights[Manager] = managerRights;
        }
    }
}
